cleanlab/rank.py

```diff
@@ -57,47 +62,56 @@ def order_label_issues(
       The columns must be ordered such that these probabilities correspond to class 0,1,2,...
       `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
 
-    rank_by : str ['normalized_margin', 'self_confidence']
+    rank_by : str ['normalized_margin', 'self_confidence', 'confidence_weighted_entropy']
       Method to order label error indices (instead of a bool mask), either:
         'normalized_margin' := normalized margin (p(label = k) - max(p(label != k)))
         'self_confidence' := [pred_probs[i][labels[i]] for i in label_issues_idx]
+        'confidence_weighted_entropy' := entropy(pred_probs) / self_confidence
+
+    rank_by_kwargs : dict
+      Optional keyword arguments to pass into `score_label_quality()` method.
+      Accepted args include:
+        adj_pred_probs : bool, default = False
 
     Returns
     -------
-      label_issues_idx : np.array (int)
-        Return the index integers of the label issues, ordered by
-        the normalized margin."""
+    label_issues_idx : np.array (int)
+      Return the index integers of the label issues, ordered by the label-quality scoring method
+      passed to rank_by.
+
+    """
 
     assert len(pred_probs) == len(labels)
+
     # Convert bool mask to index mask
     label_issues_idx = np.arange(len(labels))[label_issues_mask]
-    pred_probs_er, labels_er = pred_probs[label_issues_mask], labels[label_issues_mask]
-    if rank_by == "self_confidence":
-        label_quality_scores = get_self_confidence_for_each_label(labels_er, pred_probs_er)
-    elif rank_by == "normalized_margin":
-        label_quality_scores = get_normalized_margin_for_each_label(labels_er, pred_probs_er)
-    else:
-        raise ValueError(
-            'rank_by must be "self_confidence" or "normalized_margin", ' 'but is "' + rank_by + '".'
-        )
-    return label_issues_idx[np.argsort(label_quality_scores)]
 
+    # Calculate label quality scores
+    label_quality_scores = score_label_quality(labels, pred_probs, method=rank_by, **rank_by_kwargs)
 
-def get_self_confidence_for_each_label(
-    labels,
-    pred_probs,
-):
-    """Returns the "self-confidence" for every example in the dataset associated pred_probs
-    and labels. The self-confidence is the holdout probability that an example belongs to
-    its given class label.
+    # Get label quality scores for label issues
+    label_quality_scores_issues = label_quality_scores[label_issues_mask]
+
+    return label_issues_idx[np.argsort(label_quality_scores_issues)]
+
+
+def score_label_quality(
+    labels: np.array,
+    pred_probs: np.array,
+    method: str = "self_confidence",
```

Comment:
Why does default here not match default for `order_label_issues`?  Shouldn't we be consistent throughout?

@cgnorthcutt

Reply:
Per discussion on slack, changed default to `normalized_margin` everywhere.